eth_common/eth_types.nim

```diff
@@ -126,8 +126,11 @@ else:
   template toBlockNumber*(n: SomeInteger): BlockNumber =
     u256(n)
 
-func blockHash*(h: BlockHeader): KeccakHash =
-  assert false, "not implemented yet"
+proc toBlockNonce*(n: uint64): BlockNonce =
+  bigEndian64(addr result[0], unsafeAddr n)
+
+proc toUint*(n: BlockNonce): uint64 =
+  bigEndian64(addr result, unsafeAddr n[0])
```

Comment:
Are you sure we need to convert those to 256-bit? In the Ethash PoW algorithm nonces are always uint64: https://github.com/status-im/nim-ethash/blob/master/src/proof_of_work.nim#L147-L153

And https://github.com/ethereum/wiki/wiki/Ethash#main-loop
> nonce is the eight bytes of a 64 bit unsigned integer in big-endian order.

Reply:
Scratch my remark on nonce size, it's correct I misread >_>